<!DOCTYPE html>
<html lang="zh-cn">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <title> 生物信息学中的 SAM 格式介绍 | BIO-SPRING</title>
    
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@xiee/utils@v1.13.54/css/article.min.css">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@xiee/utils/css/heading-anchor.min.css">
    <link rel="stylesheet" href="/bio-spring/css/style.css" />
    <link rel="stylesheet" href="/bio-spring/css/fonts.css" />
    <link rel="stylesheet" href="/bio-spring/css/custom.css" />
    
  <meta name="twitter:card" content="summary">
  <meta name="twitter:title" content="生物信息学中的 SAM 格式介绍">
  <meta name="twitter:description" content="Following is a brief description of the SAM format as output by bowtie when the -S/–sam option is specified. For more details, see the SAM format specification.
When -S/–sam is specified, bowtie prints a SAM header with @HD, @SQ and @PG lines. When one or more –sam-RG arguments are specified, bowtie will also print an @RG line that includes all user-specified –sam-RG tokens separated by tabs.
Each subsequnt line corresponds to a read or an alignment. Each line is a collection of at least 12 fields separated by tabs; from left to right, the fields are:">


  </head>

  <body>

    <nav class="menu sticky-top">
    <ul>
      <li class="left">
        <a href="/"><span>BIO-SPRING</span></a>
      </li>
      
      <li>
        <a href="/bio-spring">Home</a>
      </li>
      
      <li>
        <a href="/bio-spring/post/">Blog Post</a>
      </li>
      
      <li>
        <a href="/bio-spring/publication/full-publication-list/">Publication</a>
      </li>
      
      <li>
        <a href="/bio-spring/work/">Work</a>
      </li>
      
      <li id="menu-search">
        <a href="/bio-spring/#">Search</a>
      </li>
      
    </ul>
    </nav>


<div class="container single">
<main>

<div class="article-meta">
<h1><span class="title">生物信息学中的 SAM 格式介绍</span></h1>

<h3 class="author">
gaoch
</h3>

<h3 class="date">2012-08-16</h3>
<p class="terms">
  
  
  Categories: <a href="/categories/%E5%85%B6%E5%AE%83">其它</a> <a href="/categories/%E4%BF%A1%E6%81%AF%E6%8A%80%E6%9C%AF">信息技术</a> <a href="/categories/%E7%94%9F%E7%89%A9%E5%AD%A6">生物学</a> 
  
  
  
  Tags: <a href="/tags/%E6%97%A7%E6%96%87">旧文</a> <a href="/tags/wordpress">WordPress</a> <a href="/tags/%E8%BD%AC%E8%BD%BD">转载</a> <a href="/tags/sam">SAM</a> 
  
  
</p>
</div>

<div class="article">
<p>Following is a brief description of the
<a href="http://samtools.sourceforge.net/SAM1.pdf">SAM</a> format as output by
<strong>bowtie</strong> when the <strong>-S</strong>/<strong>–sam</strong> option is specified. For more
details, see the <a href="http://samtools.sourceforge.net/SAM1.pdf">SAM format
specification</a>.</p>
<p>When <strong>-S</strong>/<strong>–sam</strong> is specified, <strong>bowtie</strong> prints a SAM header with
<strong>@HD</strong>, <strong>@SQ</strong> and <strong>@PG</strong> lines. When one or more <strong>–sam-RG</strong>
arguments are specified, <strong>bowtie</strong> will also print an <strong>@RG</strong> line that
includes all user-specified <strong>–sam-RG</strong> tokens separated by tabs.</p>
<p>Each subsequnt line corresponds to a read or an alignment. Each line is
a collection of at least 12 fields separated by tabs; from left to
right, the fields are:</p>
<ol>
<li>
<p>Name of read that aligned</p>
</li>
<li>
<p>Sum of all applicable flags. Flags relevant to Bowtie are:</p>
<table>
  <thead>
      <tr>
          <th></th>
          <th></th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td><strong>1</strong></td>
          <td>The read is one of a pair</td>
      </tr>
      <tr>
          <td><strong>2</strong></td>
          <td>The alignment is one end of a proper paired-end alignment</td>
      </tr>
      <tr>
          <td><strong>4</strong></td>
          <td>The read has no reported alignments</td>
      </tr>
      <tr>
          <td><strong>8</strong></td>
          <td>The read is one of a pair and has no reported alignments</td>
      </tr>
      <tr>
          <td><strong>16</strong></td>
          <td>The alignment is to the reverse reference strand</td>
      </tr>
      <tr>
          <td><strong>32</strong></td>
          <td>The other mate in the paired-end alignment is aligned to the reverse reference strand</td>
      </tr>
      <tr>
          <td><strong>64</strong></td>
          <td>The read is the first (#1) mate in a pair</td>
      </tr>
      <tr>
          <td><strong>128</strong></td>
          <td>The read is the second (#2) mate in a pair</td>
      </tr>
  </tbody>
</table>
<p>Thus, an unpaired read that aligns to the reverse reference strand
will have flag 16. A paired-end read that aligns and is the first
mate in the pair will have flag 83 (= 64 + 16 + 2 + 1).</p>
</li>
<li>
<p>Name of reference sequence where alignment occurs, or ordinal ID if
no name was provided</p>
</li>
<li>
<p>1-based offset into the forward reference strand where leftmost
character of the alignment occurs</p>
</li>
<li>
<p>Mapping quality</p>
</li>
<li>
<p>CIGAR string representation of alignment</p>
</li>
<li>
<p>Name of reference sequence where mate’s alignment occurs. Set to
<strong>=</strong> if the mate’s reference sequence is the same as this
alignment’s, or <strong>*</strong> if there is no mate.</p>
</li>
<li>
<p>1-based offset into the forward reference strand where leftmost
character of the mate’s alignment occurs. Offset is 0 if there is no
mate.</p>
</li>
<li>
<p>Inferred insert size. Size is negative if the mate’s alignment
occurs upstream of this alignment. Size is 0 if there is no mate.</p>
</li>
<li>
<p>Read sequence (reverse-complemented if aligned to the reverse
strand)</p>
</li>
<li>
<p>ASCII-encoded read qualities (reverse-complemented if the read
aligned to the reverse strand). The encoded quality values are on
the <a href="http://en.wikipedia.org/wiki/FASTQ_format#Variations">Phred
quality</a> scale
and the encoding is ASCII-offset by 33 (ASCII char <strong>!</strong>), similarly
to a <a href="http://en.wikipedia.org/wiki/FASTQ_format">FASTQ</a> file.</p>
</li>
<li>
<p>Optional fields. Fields are tab-separated. For descriptions of all
possible optional fields, see the SAM format specification.
<strong>bowtie</strong>outputs some of these optional fields for each alignment,
depending on the type of the alignment:</p>
</li>
</ol>
<ul>
<li>NM:i:&lt;N&gt; Aligned read has an edit distance of &lt;N&gt;.</li>
<li>CM:i:&lt;N&gt; Aligned read has an edit distance of &lt;N&gt; in
colorspace. This field is present in addition to the NM field in
-C/–color mode, but is omitted otherwise.</li>
<li>MD:Z:&lt;S&gt; For aligned reads, &lt;S&gt; is a string
representation of the mismatched reference bases in the alignment.
See SAM format specification for details. For colorspace alignments,
&lt;S&gt; describes the decoded nucleotide alignment, not the
colorspace alignment.</li>
<li>XA:i:&lt;N&gt; Aligned read belongs to stratum &lt;N&gt;. See Strata
for definition.</li>
<li>XM:i:&lt;N&gt; For a read with no reported alignments, &lt;N&gt; is
0 if the read had no alignments. If -m was specified and the read’s
alignments were supressed because the -m ceiling was exceeded,
&lt;N&gt; equals the -m ceiling + 1, to indicate that there were at
least that many valid alignments (but all were suppressed). In -M
mode, if the alignment was randomly selected because the -M ceiling
was exceeded, &lt;N&gt; equals the -M ceiling + 1, to indicate that
there were at least that many valid alignments (of which one was
reported at random).</li>
</ul>
<p> </p>
<p>SAM文件定义</p>
<p>SAM是一种序列比对格式标准，
由sanger制定，是以TAB为分割符的文本格式。主要应用于测序序列mapping到基因组上的结果表示，当然也可以表示任意的多重比对结果。</p>
<p>不同的软件，不同的时期，不同的研究方向，都会创建一种或者多种格式标准，当然根据当时的需要，创建符合需求的标准，也是最容易的事情，而反过来想
要真正的理解标准，也必须理解为什么要创建这样的标准，解决什么样的需要。我前面的有篇文章已经对于现有的多重比对的格式进行总结，但其更多的站在比较基
因组学的角度。当我们去了解sam标准格式是什么的时候，就要思考既然以及有了这么多得标准，为什么还要定义SAM标准，当然拿所有的格式进行比较也并非
易事，但是简单的对比，就可以了解其中一二，比如aln格式，是比对视图化的展示，存储的信息不够结构化，无法方便的作为另外程序的输入；表示信息的有限
性，如果100个多重比对序列放到一个文件中，查看维护就会非常困难；还有些格式标准挺强大，但是太繁琐，同时不够灵活。那么反过来就是SAM格式的优
点，那么SAM如何做到这一点的呢？</p>
<p>SAM要处理好的问题：</p>
<ul>
<li>非常多序列（read)，mapping到多个参考基因组（reference）上；</li>
<li>同一条序列，分多段（segment）比对到参考基因组上；</li>
<li>无限量的，结构化信息表示，包括错配、删除、插入等比对信息；</li>
</ul>
<p>SAM分为两部分，注释信息（header section）和比对结果部分（alignment
section），注释信息可有可无，都是以@开头，用不同的tag表示不同的信息，主要有@HD，说明符合标准的版本、对比序列的排列顺序；@SQ，参
考序列说明；@RG，比对上的序列（read）说明；@PG，使用的程序说明；@CO，任意的说明信息。</p>
<p><a href="http://boyun.sh.cn/bio/wp-content/uploads/2012/07/btp352f1.jpg"><img src="http://boyun.sh.cn/bio/wp-content/uploads/2012/07/btp352f1.jpg" title="SAM格式示例" class="alignnone size-full wp-image-1891" width="447" height="360" alt="SAM格式示例" /></a></p>
<p>比对结果部分（alignment
section），每一行表示一个片段（segment）的比对信息，包括11个必须的字段（mandatory
fields）和一个可选的字段，字段之间用tag分割。必须的字段有11个，顺序固定，不可用时，根据字段定义，可以为’0‘或者’*‘，这是11个字
段包括：</p>
<ol>
<li>QNAME，比对片段的（template）的编号；</li>
<li>FLAG，位标识，template
mapping情况的数字表示，每一个数字代表一种比对情况，这里的值是符合情况的数字相加总和；</li>
<li>RNAME，参考序列的编号，如果注释中对SQ-SN进行了定义，这里必须和其保持一致，另外对于没有mapping上的序列，这里是’*‘；</li>
<li>POS，比对上的位置，注意是从1开始计数，没有比对上，此处为0；</li>
<li>MAPQ，mappint的质量；</li>
<li>CIGAR，简要比对信息表达式（Compact Idiosyncratic Gapped Alignment
Report），其以参考序列为基础，使用数字加字母表示比对结果，比如3S6M1P1I4M，前三个碱基被剪切去除了，然后6个比对上了，然后打开了一
个缺口，有一个碱基插入，最后是4个比对上了，是按照顺序的；</li>
<li>RNEXT，下一个片段比对上的参考序列的编号，没有另外的片段，这里是’*‘，同一个片段，用’=‘；</li>
<li>PNEXT，下一个片段比对上的位置，如果不可用，此处为0；</li>
<li>TLEN，Template的长度，最左边得为正，最右边的为负，中间的不用定义正负，不分区段（single-segment)的比对上，或者不可用时，此处为0；</li>
<li>SEQ，序列片段的序列信息，如果不存储此类信息，此处为’*‘，注意CIGAR中M/I/S/=/X对应数字的和要等于序列长度；</li>
<li>QUAL，序列的质量信息，格式同FASTQ一样。</li>
</ol>
<p>可选字段（optional
fields)，格式如：TAG:TYPE:VALUE，其中TAG有两个大写字母组成，每个TAG代表一类信息，每一行一个TAG只能出现一次，TYPE表示TAG对应值的类型，可以是字符串、整数、字节、数组等。</p>
<p>要注意的几个概念，以及与之对应的模型：</p>
<ul>
<li>reference</li>
<li>read</li>
<li>segment</li>
<li>template（参考序列和比对上的序列共同组成的序列为template）</li>
<li>alignment</li>
<li>seq</li>
</ul>
<p> </p>
<p><a href="http://bowtie-bio.sourceforge.net/manual.shtml">http://bowtie-bio.sourceforge.net/manual.shtml</a>#sam-bowtie-output</p>
<p><a href="http://boyun.sh.cn/bio/?p=1890">http://boyun.sh.cn/bio/?p=1890</a></p>

</div>
</main>

<section class="appendix">


<div>
  <div class="side side-left"><h3>作者简介</h3></div>
  <div>





<div>Chun-Hui Gao is a Research Associate at <a href="http://www.hzau.edu.cn">Huazhong Agricultural University</a>.</div>


</div>
</div>






<div>
  <div class="side side-left"><h3>重复使用</h3></div>
  Text and figures are licensed under <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution CC BY 4.0</a>. The source code is licensed under MIT. The full source is available at <a href="https://github.com/yihui/hugo-prose">https://github.com/yihui/hugo-prose</a>.
</div>



<div>
  <div class="side side-left"><h3>欢迎修订</h3></div>
  
  
  
    
    
  
  如果您发现本文里含有任何错误（包括错别字和标点符号），欢迎<a href="https://github.com/gaospecial/bio-spring/edit/master/content/post/2012-08-16-sam-format/index.zh.md" id="edit-link">在本站的 GitHub 项目里提交修订意见。</a>
</div>




<div>
  <div class="side side-left"><h3>引用本文</h3></div>
  <p>如果您使用了本文的内容，请按照以下方式引用：</p>
  <pre><code>gaoch (2012). 生物信息学中的 SAM 格式介绍. BIO-SPRING. bio-spring/post/2012/08/16/2012-08-16-sam-format/</code></pre>
  <p>BibTeX citation</p>
  <pre><code>@misc{
  title = "生物信息学中的 SAM 格式介绍",
  author = "gaoch",
  year = "2012",
  journal = "BIO-SPRING",
  note = "bio-spring/post/2012/08/16/2012-08-16-sam-format/"
}</code></pre>
</div>

</section>



<nav class="post-nav">
  <span class="nav-next">&larr; <a href="bio-spring/post/2012/06/05/2012-06-05-matlab-chinamap-coloring/" title=下一篇&#32;(旧)>MatLab中国地图进阶：以不同的颜色表示各省数据的差异</a></span>
  &hercon;
  <span class="nav-prev"><a href="bio-spring/post/2012/08/20/2012-08-20-ubuntu-bowtie/" title=上一篇&#32;(新)>在Ubuntu环境中设置Bowtie</a> &rarr;</span>
</nav>




<section class="comments">
  <div id="disqus_thread"></div>
  <script>
  var disqus_config = function () {
  
  };
  (function() {
    var inIFrame = function() {
      var iframe = true;
      try { iframe = window.self !== window.top; } catch (e) {}
      return iframe;
    };
    if (inIFrame()) return;
    var disqus_js = '//bio-spring.disqus.com/embed.js';
    var d = document, s = d.createElement('script');
    s.src = disqus_js; s.async = true;
    s.setAttribute('data-timestamp', +new Date());
    var t = d.getElementById('disqus_thread');
    var b = false, l = function(scroll) {
      if (b) return;
      (d.head || d.body).appendChild(s); b = true;
      if (scroll) t.scrollIntoView();
    }
    s.onerror = function(e) {
      if (sessionStorage.getItem('failure-note')) return;
      t.innerText = 'Sorry, but you cannot make comments because Disqus failed to load for some reason. It is known to be blocked in China. If you are sure it is not blocked in your region, please refresh the page. 中国大陆地区读者需要翻墙才能发表评论。';
      t.style.border = '1px dashed';
      t.style.padding = '.5em';
      t.style.background = 'lightyellow';
      sessionStorage.setItem('failure-note', true);
    };
    
    if (location.hash.match(/^#comment-[0-9]+$/)) return l(true);
    var c = function() {
      if (b) return;
      var rect = t.getBoundingClientRect();
      if (rect.top < window.innerHeight && rect.bottom >= 0) l();
    };
    window.addEventListener('load', c);
    d.addEventListener('scroll', c);
  })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</section>


</div>


  <footer>
  
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.6.2"></script>





      <script async src="https://www.googletagmanager.com/gtag/js?id=G-77M8KZTCMR"></script>
      <script>
        var doNotTrack = false;
        if ( false ) {
          var dnt = (navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack);
          var doNotTrack = (dnt == "1" || dnt == "yes");
        }
        if (!doNotTrack) {
          window.dataLayer = window.dataLayer || [];
          function gtag(){dataLayer.push(arguments);}
          gtag('js', new Date());
          gtag('config', 'G-77M8KZTCMR');
        }
      </script>


<script src="https://cdn.jsdelivr.net/combine/npm/@xiee/utils/js/number-sections.min.js,npm/@xiee/utils/js/toc.min.js,npm/@xiee/utils/js/toc-highlight.min.js,npm/@xiee/utils/js/sidenotes.min.js,npm/@xiee/utils/js/right-quote.min.js,npm/@xiee/utils/js/center-img.min.js,npm/@xiee/utils/js/fix-pandoc.min.js,npm/@xiee/utils/js/heading-anchor.min.js" defer></script>




  <div class="footer">
  
  <ul class="copyright">
    <li>© <a href="https://bio-spring.top">Bio-Spring</a> 2006 &ndash; 2025 <a href="https://beian.miit.gov.cn">豫ICP备2021003240号</a></li>
  </ul>
  
  <ul>
    
    <li>
      <a href="/bio-spring/404.html">Contact</a>
    </li>
    
    <li class="optional">
      <a href="/bio-spring/categories/">Categories</a>
    </li>
    
    <li class="optional">
      <a href="/bio-spring/tags/">Tags</a>
    </li>
    
    <li id="menu-edit">
      <a href="#">Suggest an edit</a>
    </li>
    
    <li>
      <a href="#">Back to top</a>
    </li>
    
  </ul>
  </div>
  
  </footer>
  <script src="/bio-spring/js/features.js" defer></script>
  </body>
</html>

