<!DOCTYPE html>
<html lang="zh-cn">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <title> MetaPhlAn：利用宏基因组分析种群结构 | BIO-SPRING</title>
    
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@xiee/utils@v1.13.54/css/article.min.css">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@xiee/utils/css/heading-anchor.min.css">
    <link rel="stylesheet" href="/bio-spring/css/style.css" />
    <link rel="stylesheet" href="/bio-spring/css/fonts.css" />
    <link rel="stylesheet" href="/bio-spring/css/custom.css" />
    
  <meta name="twitter:card" content="summary">
  <meta name="twitter:title" content="MetaPhlAn：利用宏基因组分析种群结构">
  <meta name="twitter:description" content="宏基因组中包含了环境中各种微生物的基因组信息，理论上可以得出种群结构。该如何利用宏基因组高通量测序数据分析种群结构呢？是不是要从中提取16S rRNA的数据， 组装之后使用Mothur或QIIME分析呢？
MetaPhlAn有一个不一样的想法。
MetaPhlAn is a computational tool for profiling the composition of microbial communities from metagenomic shotgun sequencing data. MetaPhlAn relies on unique clade-specific marker genes identified from 3,000 reference genomes, allowing:
up to 25,000 reads-per-second (on one CPU) analysis speed (orders of magnitude faster compared to existing methods); unambiguous taxonomic assignments as the MetaPhlAn markers are clade-specific; accurate estimation of organismal relative abundance (in terms of number of cells rather than fraction of reads); species-level resolution for bacterial and archaeal organisms; extensive validation of the profiling accuracy on several synthetic datasets and on thousands of real metagenomes.">


  </head>

  <body>

    <nav class="menu sticky-top">
    <ul>
      <li class="left">
        <a href="/"><span>BIO-SPRING</span></a>
      </li>
      
      <li>
        <a href="/bio-spring">Home</a>
      </li>
      
      <li>
        <a href="/bio-spring/post/">Blog Post</a>
      </li>
      
      <li>
        <a href="/bio-spring/publication/full-publication-list/">Publication</a>
      </li>
      
      <li>
        <a href="/bio-spring/work/">Work</a>
      </li>
      
      <li id="menu-search">
        <a href="/bio-spring/#">Search</a>
      </li>
      
    </ul>
    </nav>


<div class="container single">
<main>

<div class="article-meta">
<h1><span class="title">MetaPhlAn：利用宏基因组分析种群结构</span></h1>

<h3 class="author">
gaoch
</h3>

<h3 class="date">2016-03-20</h3>
<p class="terms">
  
  
  Categories: <a href="/categories/%E5%85%B6%E5%AE%83">其它</a> <a href="/categories/%E4%BF%A1%E6%81%AF%E6%8A%80%E6%9C%AF">信息技术</a> <a href="/categories/%E7%94%9F%E7%89%A9%E5%AD%A6">生物学</a> 
  
  
  
  Tags: <a href="/tags/%E6%97%A7%E6%96%87">旧文</a> <a href="/tags/wordpress">WordPress</a> <a href="/tags/metaphlan">MetaPhlAn</a> <a href="/tags/mothur">mothur</a> <a href="/tags/qiime">QIIME</a> 
  
  
</p>
</div>

<div class="article">
<p>宏基因组中包含了环境中各种微生物的基因组信息，理论上可以得出种群结构。该如何利用宏基因组高通量测序数据分析种群结构呢？是不是要从中提取16S
rRNA的数据，
组装之后使用<a href="http://bio-spring.top/mothur-and-qiime/">Mothur或QIIME</a>分析呢？</p>
<p><a href="http://huttenhower.sph.harvard.edu/metaphlan">MetaPhlAn</a>有一个不一样的想法。</p>
<blockquote>
<p>MetaPhlAn is a computational tool for profiling the composition of
microbial communities from metagenomic shotgun sequencing data.
MetaPhlAn relies on <strong>unique clade-specific marker genes</strong> identified
from 3,000 reference genomes, allowing:</p>
<ul>
<li>up to 25,000 reads-per-second (on one CPU) analysis speed (orders
of magnitude faster compared to existing methods);</li>
<li>unambiguous taxonomic assignments as the MetaPhlAn markers are
clade-specific;</li>
<li>accurate estimation of organismal relative abundance (in terms of
number of cells rather than fraction of reads);</li>
<li>species-level resolution for bacterial and archaeal organisms;</li>
<li>extensive validation of the profiling accuracy on several
synthetic datasets and on thousands of real metagenomes.</li>
</ul>
<p> </p>
<p>它使用<strong>物种特异性的标记基因</strong>来进行分析，并且能够得到很好的种群分析结果。</p>
<p> </p></blockquote>

</div>
</main>

<section class="appendix">


<div>
  <div class="side side-left"><h3>作者简介</h3></div>
  <div>





<div>Chun-Hui Gao is a Research Associate at <a href="http://www.hzau.edu.cn">Huazhong Agricultural University</a>.</div>


</div>
</div>






<div>
  <div class="side side-left"><h3>重复使用</h3></div>
  Text and figures are licensed under <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution CC BY 4.0</a>. The source code is licensed under MIT. The full source is available at <a href="https://github.com/yihui/hugo-prose">https://github.com/yihui/hugo-prose</a>.
</div>



<div>
  <div class="side side-left"><h3>欢迎修订</h3></div>
  
  
  
    
    
  
  如果您发现本文里含有任何错误（包括错别字和标点符号），欢迎<a href="https://github.com/gaospecial/bio-spring/edit/master/content/post/2016-03-20-metaphlan-metagenomic-phylogenetic-analysis/index.zh.md" id="edit-link">在本站的 GitHub 项目里提交修订意见。</a>
</div>




<div>
  <div class="side side-left"><h3>引用本文</h3></div>
  <p>如果您使用了本文的内容，请按照以下方式引用：</p>
  <pre><code>gaoch (2016). MetaPhlAn：利用宏基因组分析种群结构. BIO-SPRING. bio-spring/post/2016/03/20/2016-03-20-metaphlan-metagenomic-phylogenetic-analysis/</code></pre>
  <p>BibTeX citation</p>
  <pre><code>@misc{
  title = "MetaPhlAn：利用宏基因组分析种群结构",
  author = "gaoch",
  year = "2016",
  journal = "BIO-SPRING",
  note = "bio-spring/post/2016/03/20/2016-03-20-metaphlan-metagenomic-phylogenetic-analysis/"
}</code></pre>
</div>

</section>



<nav class="post-nav">
  <span class="nav-next">&larr; <a href="bio-spring/post/2016/02/14/letter-2/" title=下一篇&#32;(旧)>Letter 2: On Discursiveness in Reading</a></span>
  &hercon;
  <span class="nav-prev"><a href="bio-spring/post/2016/03/22/2016-03-22-drawing-conical-flask-with-powerpoint/" title=上一篇&#32;(新)>作图神器PowerPoint：三角瓶</a> &rarr;</span>
</nav>




<section class="comments">
  <div id="disqus_thread"></div>
  <script>
  var disqus_config = function () {
  
  };
  (function() {
    var inIFrame = function() {
      var iframe = true;
      try { iframe = window.self !== window.top; } catch (e) {}
      return iframe;
    };
    if (inIFrame()) return;
    var disqus_js = '//bio-spring.disqus.com/embed.js';
    var d = document, s = d.createElement('script');
    s.src = disqus_js; s.async = true;
    s.setAttribute('data-timestamp', +new Date());
    var t = d.getElementById('disqus_thread');
    var b = false, l = function(scroll) {
      if (b) return;
      (d.head || d.body).appendChild(s); b = true;
      if (scroll) t.scrollIntoView();
    }
    s.onerror = function(e) {
      if (sessionStorage.getItem('failure-note')) return;
      t.innerText = 'Sorry, but you cannot make comments because Disqus failed to load for some reason. It is known to be blocked in China. If you are sure it is not blocked in your region, please refresh the page. 中国大陆地区读者需要翻墙才能发表评论。';
      t.style.border = '1px dashed';
      t.style.padding = '.5em';
      t.style.background = 'lightyellow';
      sessionStorage.setItem('failure-note', true);
    };
    
    if (location.hash.match(/^#comment-[0-9]+$/)) return l(true);
    var c = function() {
      if (b) return;
      var rect = t.getBoundingClientRect();
      if (rect.top < window.innerHeight && rect.bottom >= 0) l();
    };
    window.addEventListener('load', c);
    d.addEventListener('scroll', c);
  })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</section>


</div>


  <footer>
  
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.6.2"></script>





      <script async src="https://www.googletagmanager.com/gtag/js?id=G-77M8KZTCMR"></script>
      <script>
        var doNotTrack = false;
        if ( false ) {
          var dnt = (navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack);
          var doNotTrack = (dnt == "1" || dnt == "yes");
        }
        if (!doNotTrack) {
          window.dataLayer = window.dataLayer || [];
          function gtag(){dataLayer.push(arguments);}
          gtag('js', new Date());
          gtag('config', 'G-77M8KZTCMR');
        }
      </script>


<script src="https://cdn.jsdelivr.net/combine/npm/@xiee/utils/js/number-sections.min.js,npm/@xiee/utils/js/toc.min.js,npm/@xiee/utils/js/toc-highlight.min.js,npm/@xiee/utils/js/sidenotes.min.js,npm/@xiee/utils/js/right-quote.min.js,npm/@xiee/utils/js/center-img.min.js,npm/@xiee/utils/js/fix-pandoc.min.js,npm/@xiee/utils/js/heading-anchor.min.js" defer></script>




  <div class="footer">
  
  <ul class="copyright">
    <li>© <a href="https://bio-spring.top">Bio-Spring</a> 2006 &ndash; 2025 <a href="https://beian.miit.gov.cn">豫ICP备2021003240号</a></li>
  </ul>
  
  <ul>
    
    <li>
      <a href="/bio-spring/404.html">Contact</a>
    </li>
    
    <li class="optional">
      <a href="/bio-spring/categories/">Categories</a>
    </li>
    
    <li class="optional">
      <a href="/bio-spring/tags/">Tags</a>
    </li>
    
    <li id="menu-edit">
      <a href="#">Suggest an edit</a>
    </li>
    
    <li>
      <a href="#">Back to top</a>
    </li>
    
  </ul>
  </div>
  
  </footer>
  <script src="/bio-spring/js/features.js" defer></script>
  </body>
</html>

