<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <title> MetaPhlAn：利用宏基因组分析种群结构 | Hugo Prose</title>
    
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/combine/npm/@xiee/utils/css/article.min.css,npm/@xiee/utils/css/heading-anchor.min.css">
    <link rel="stylesheet" href="/css/style.css" />
    <link rel="stylesheet" href="/css/fonts.css" />
    <link rel="stylesheet" href="/css/custom.css" />
    <meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="MetaPhlAn：利用宏基因组分析种群结构"/>
<meta name="twitter:description" content="宏基因组中包含了环境中各种微生物的基因组信息，理论上可以得出种群结构。该如何利用宏基因组高通量测序数据分析种群结构呢？是不是要从中提取16S rRNA的数据， 组装之后使用Mothur或QIIME分析呢？
MetaPhlAn有一个不一样的想法。
MetaPhlAn is a computational tool for profiling the composition of microbial communities from metagenomic shotgun sequencing data. MetaPhlAn relies on unique clade-specific marker genes identified from 3,000 reference genomes, allowing:
up to 25,000 reads-per-second (on one CPU) analysis speed (orders of magnitude faster compared to existing methods); unambiguous taxonomic assignments as the MetaPhlAn markers are clade-specific; accurate estimation of organismal relative abundance (in terms of number of cells rather than fraction of reads); species-level resolution for bacterial and archaeal organisms; extensive validation of the profiling accuracy on several synthetic datasets and on thousands of real metagenomes."/>


  </head>

  <body>

    <nav class="menu">
    <ul>
      <li class="left">
        <a href="/"><span>Hugo Prose</span></a>
      </li>
      
      <li>
        <a href="/">Home</a>
      </li>
      
      <li>
        <a href="/about/">About</a>
      </li>
      
      <li>
        <a href="/sitemap/">Sitemap</a>
      </li>
      
      <li id="menu-search">
        <a href="/#">Search</a>
      </li>
      
      <li>
        <a href="/index.xml">Subscribe</a>
      </li>
      
    </ul>
    </nav>


<div class="container single">
<main>

<div class="article-meta">
<h1><span class="title">MetaPhlAn：利用宏基因组分析种群结构</span></h1>

<h3 class="date">2016-03-20</h3>
<p class="terms">
  
  
  
  
  Tags: <a href="/tags/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF-bioinformatics">生物信息 BIOINFORMATICS</a> <a href="/tags/%E7%94%9F%E7%89%A9%E8%BD%AF%E4%BB%B6-bio-software">生物软件 BIO-SOFTWARE</a> <a href="/tags/metaphlan">MetaPhlAn</a> <a href="/tags/mothur">mothur</a> <a href="/tags/qiime">QIIME</a> 
  
  
</p>
</div>

<div class="article">
<p>宏基因组中包含了环境中各种微生物的基因组信息，理论上可以得出种群结构。该如何利用宏基因组高通量测序数据分析种群结构呢？是不是要从中提取16S
rRNA的数据，
组装之后使用<a href="http://bio-spring.top/mothur-and-qiime/">Mothur或QIIME</a>分析呢？</p>
<p><a href="http://huttenhower.sph.harvard.edu/metaphlan">MetaPhlAn</a>有一个不一样的想法。</p>
<blockquote>
<p>MetaPhlAn is a computational tool for profiling the composition of
microbial communities from metagenomic shotgun sequencing data.
MetaPhlAn relies on <strong>unique clade-specific marker genes</strong> identified
from 3,000 reference genomes, allowing:</p>
<ul>
<li>up to 25,000 reads-per-second (on one CPU) analysis speed (orders
of magnitude faster compared to existing methods);</li>
<li>unambiguous taxonomic assignments as the MetaPhlAn markers are
clade-specific;</li>
<li>accurate estimation of organismal relative abundance (in terms of
number of cells rather than fraction of reads);</li>
<li>species-level resolution for bacterial and archaeal organisms;</li>
<li>extensive validation of the profiling accuracy on several
synthetic datasets and on thousands of real metagenomes.</li>
</ul>
<p> </p>
<p>它使用<strong>物种特异性的标记基因</strong>来进行分析，并且能够得到很好的种群分析结果。</p>
<p> </p>
</blockquote>

</div>
</main>

<section class="appendix">







<div>
  <div class="side side-left"><h3>Reuse</h3></div>
  Text and figures are licensed under <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution CC BY 4.0</a>. The source code is licensed under MIT. The full source is available at <a href="https://github.com/yihui/hugo-prose">https://github.com/yihui/hugo-prose</a>.
</div>



<div>
  <div class="side side-left"><h3>Suggest changes</h3></div>
  
  
  
    
    
  
  If you find any mistakes (including typos) or want to suggest changes, please feel free to <a href="https://github.com/yihui/hugo-prose/edit/master/exampleSite/content/post/2016-03-20-metaphlan-metagenomic-phylogenetic-analysis.md" id="edit-link">edit the source file of this page on Github and create a pull request.</a>
</div>




</section>



<nav class="post-nav">
  <span class="nav-next">&larr; <a href="/post/2016/02/14/letter-2-on-discursiveness-in-reading/" title=Next&#32;post&#32;(older)>Letter 2: On Discursiveness in Reading</a></span>
  &hercon;
  <span class="nav-prev"><a href="/post/2016/03/22/2016-03-22-drawing-conical-flask-with-powerpoint/" title=Previous&#32;post&#32;(newer)>作图神器PowerPoint：三角瓶</a> &rarr;</span>
</nav>





</div>


  <footer>
  






<script src="https://cdn.jsdelivr.net/combine/npm/@xiee/utils/js/number-sections.min.js,npm/@xiee/utils/js/toc.min.js,npm/@xiee/utils/js/sidenotes.min.js,npm/@xiee/utils/js/right-quote.min.js,npm/@xiee/utils/js/center-img.min.js,npm/@xiee/utils/js/fix-pandoc.min.js,npm/@xiee/utils/js/heading-anchor.min.js" defer></script>




  <div class="footer">
  
  <ul class="copyright">
    <li>© <a href="https://en.wikipedia.org/wiki/Seneca_the_Younger">Lucius Annaeus Seneca</a> 4 BC &ndash; AD 65</li>
  </ul>
  
  <ul>
    
    <li>
      <a href="/404.html">Contact</a>
    </li>
    
    <li class="optional">
      <a href="/categories/">Categories</a>
    </li>
    
    <li class="optional">
      <a href="/tags/">Tags</a>
    </li>
    
    <li id="menu-edit">
      <a href="#">Suggest an edit</a>
    </li>
    
    <li>
      <a href="#">Back to top</a>
    </li>
    
  </ul>
  </div>
  
  </footer>
  <script src="/js/features.js" defer></script>
  </body>
</html>

