<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <title> A Genbank –&gt; GTF converter derived from BioPerl | Hugo Prose</title>
    
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/combine/npm/@xiee/utils/css/article.min.css,npm/@xiee/utils/css/heading-anchor.min.css">
    <link rel="stylesheet" href="/css/style.css" />
    <link rel="stylesheet" href="/css/fonts.css" />
    <link rel="stylesheet" href="/css/custom.css" />
    <meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="A Genbank –&gt; GTF converter derived from BioPerl"/>
<meta name="twitter:description" content="BioPerl has a Genbank –&gt; GFF converter (the script is bp_genbank2gff3.pl), however, this converter generated GFF/GTF file could not work in Bioconductor when using the function maketxdbfromGFF to construct a transcript database.
Currently, I have developed such a converter that may extract information from a NCBI Genbank file and write them to GFF version 2, or GTF format. This script is derived from BioPerl, so you need firstly get the module installed in your system."/>


  </head>

  <body>

    <nav class="menu">
    <ul>
      <li class="left">
        <a href="/"><span>Hugo Prose</span></a>
      </li>
      
      <li>
        <a href="/">Home</a>
      </li>
      
      <li>
        <a href="/about/">About</a>
      </li>
      
      <li>
        <a href="/sitemap/">Sitemap</a>
      </li>
      
      <li id="menu-search">
        <a href="/#">Search</a>
      </li>
      
      <li>
        <a href="/index.xml">Subscribe</a>
      </li>
      
    </ul>
    </nav>


<div class="container single">
<main>

<div class="article-meta">
<h1><span class="title">A Genbank –&gt; GTF converter derived from BioPerl</span></h1>

<h3 class="date">2015-10-14</h3>
<p class="terms">
  
  
  
  
  Tags: <a href="/tags/bioperl-bioperl">BioPerl BIOPERL</a> <a href="/tags/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF-bioinformatics">生物信息 BIOINFORMATICS</a> <a href="/tags/bioperl">BioPerl</a> <a href="/tags/genbank">GenBank</a> <a href="/tags/gff">GFF</a> <a href="/tags/gtf">GTF</a> 
  
  
</p>
</div>

<div class="article">
<p>BioPerl has a Genbank –&gt; GFF converter (the script is
bp_genbank2gff3.pl), however, this converter generated GFF/GTF file
could not work in Bioconductor when using the function
<strong>maketxdbfromGFF</strong> to construct a transcript database.</p>
<p>Currently, I have developed such a converter that may extract
information from a NCBI Genbank file and write them to GFF version 2, or
GTF format. This script is derived from BioPerl, so you need firstly
<a href="http://bio-spring.top/reinstall-bioperl/">get the module installed</a> in
your system. To check whether BioPerl is ready, just refer to this post:
<a href="http://bio-spring.top/check-bioperl/">http://bio-spring.top/check-bioperl/</a> .</p>
<p>Here is the synopsis of this script.</p>
<h2 id="synopsis">SYNOPSIS</h2>
<p><code>     NCBI_genbank2gtf.pl [options] filename</code></p>
<p>Options:<br>
&ndash;help -h display this message<br>
&ndash;input -i NCBI GenBank file<br>
&ndash;out -o filename of GTF output</p>
<p>Examples:</p>
<p>perl NCBI_genbank2gtf.pl -i seq.gb -o seq.gtf<br>
perl NCBI_genbank2gtf.pl seq.gb &gt; seq.gtf</p>
<p>The codes is as follows:<br>
<code>#!/usr/bin/perl</code></p>
<p>=pod</p>
<p>=head1 NAME</p>
<p>NCBI_genbank2gtf.pl &ndash; Genbank -&gt; Bioconductor friendly GTF</p>
<p>=head1 SYNOPSIS</p>
<p>NCBI_genbank2gtf.pl [options] filename</p>
<p>Options:<br>
&ndash;help -h display this message<br>
&ndash;input -i NCBI GenBank file<br>
&ndash;out -o filename of GTF output</p>
<p>Examples:</p>
<p>perl NCBI_genbank2gtf.pl -i seq.gb -o seq.gtf<br>
perl NCBI_genbank2gtf.pl seq.gb &gt; seq.gtf</p>
<p>=head1 DESCRIPTION</p>
<p>Use this program to generate bioconductor friendly GTF files from NCBI
GenBank.<br>
The file may be used in *maketxdbfromGFF* function and should work
well.</p>
<p>=head1 AUTHOR</p>
<p>Chun-Hui, Gao (<a href="mailto:gaoch@thelifesciencecentury.com">gaoch@thelifesciencecentury.com</a>)</p>
<p>Copyright (c) 2015 <a href="https://www.thelifesciencecentury.com">www.thelifesciencecentury.com</a>.</p>
<p>=cut</p>
<p>use strict;<br>
use Data::Dumper;<br>
use Pod::Usage;<br>
use Bio::SeqIO;<br>
use Getopt::Long;</p>
<p>my ($help, $genbank_input, $gtf_output);</p>
<p>my $ok= GetOptions( &lsquo;help|h&rsquo; =&gt; \$help,<br>
&lsquo;input|i=s&rsquo; =&gt; \$genbank_input,<br>
&lsquo;out|o=s&rsquo; =&gt; \$gtf_output );<br>
pod2usage(2) if $help || !$ok;</p>
<p>$genbank_input = shift @ARGV unless ($genbank_input );</p>
<p>open *IN, &ldquo;&lt; $genbank_input &quot; or die pod2usage(2); my $out; if
($gtf_output) { open $out, &ldquo;&gt; $gtf_output&rdquo; or die &ldquo;Can&rsquo;t open file
$gtf_output:$@\n&rdquo;;<br>
}<br>
else {<br>
$out = *STDOUT;<br>
}</p>
<p>my %seen;<br>
my $in = Bio::SeqIO-&gt;new(-fh =&gt; \*IN, -format =&gt;
&ldquo;genbank&rdquo;);<br>
while ( my $seq = $in-&gt;next_seq() ) {<br>
my $seq_acc = $seq-&gt;accession_number;</p>
<p># abort if there are no features<br>
warn &ldquo;$seq_acc has no features, skipping\n&rdquo; and next<br>
if !$seq-&gt;all_SeqFeatures;</p>
<p># construct a GFF header<br>
# add: get source_type from attributes of source feature? chromosome=X
tag<br>
# also combine 1st ft line here with source ft from $seq ..<br>
my $header = gff_header($seq);<br>
print $out $header;<br>
warn &ldquo;# working on $seq_acc\n&rdquo;;</p>
<p>for my $feature ( $seq-&gt;get_all_SeqFeatures ) {</p>
<p>my $method = $feature-&gt;primary_tag;<br>
$seen{$method} ++;<br>
$feature = maptags2gff($feature);<br>
print $out $feature-&gt;gff_string(), &ldquo;\n&rdquo;;<br>
if ($method eq &lsquo;CDS&rsquo;){<br>
$feature-&gt;primary_tag(&ldquo;exon&rdquo;);<br>
$feature-&gt;add_tag_value(&ldquo;exon_number&rdquo;, 1);<br>
$feature-&gt;add_tag_value(&ldquo;exon_id&rdquo;,
$feature-&gt;get_tag_values(&ldquo;transcript_id&rdquo;));<br>
$feature = maptags2gff($feature);<br>
print $out $feature-&gt;gff_string(), &ldquo;\n&rdquo;;<br>
}</p>
<p>}</p>
<p>}</p>
<p>warn &ldquo;# Count of features in $genbank_input:\n&rdquo;;<br>
map {warn sprintf(&rdquo;# %20s:%6d\n&quot;, $_, $seen{$_});} keys %seen;</p>
<p>sub gff_header {<br>
my ($seq) = @_;<br>
my $head = sprintf(&quot;##gff-version 2\n# sequence-region: %s
(1..%d)\n&quot;, $seq-&gt;accession_number, $seq-&gt;length);<br>
$head .= sprintf( &ldquo;# %s\n&rdquo;, $seq-&gt;desc);<br>
$head .= &ldquo;# converted by NCBI_genbank2gtf.pl\n&rdquo;;</p>
<p>#~ print Dumper $acc, $desc, $end;<br>
return $head;</p>
<p>}</p>
<p>sub maptags2gff {<br>
my $f = shift;<br>
my $method = $f-&gt;primary_tag;<br>
my %TAG_MAP;<br>
if ($method eq &lsquo;gene&rsquo; || $method eq &lsquo;misc_RNA&rsquo; || $method eq &lsquo;rRNA&rsquo;
|| $method eq &rsquo;tRNA&rsquo;){<br>
$f-&gt;add_tag_value(&rsquo;transcript_id&rsquo;,$f-&gt;get_tag_values(&rsquo;locus_tag&rsquo;));<br>
%TAG_MAP = ( gene =&gt; &lsquo;gene_name&rsquo;,<br>
transcript_id =&gt; &rsquo;transcript_id&rsquo;,<br>
locus_tag =&gt; &lsquo;gene_id&rsquo; );<br>
}<br>
elsif ($method eq &lsquo;CDS&rsquo;) {<br>
$f-&gt;add_tag_value(&rsquo;transcript_id&rsquo;,$f-&gt;get_tag_values(&rsquo;locus_tag&rsquo;));<br>
%TAG_MAP = ( locus_tag =&gt; &lsquo;gene_id&rsquo;,<br>
protein_id =&gt; &lsquo;protein_id&rsquo;,<br>
transcript_id =&gt; &rsquo;transcript_id&rsquo;,<br>
gene =&gt; &lsquo;gene_name&rsquo;,<br>
product =&gt; &lsquo;product&rsquo;);</p>
<p>}<br>
elsif ($method eq &rsquo;exon&rsquo;) {</p>
<p>%TAG_MAP = ( gene_id =&gt; &lsquo;gene_id&rsquo;,<br>
protein_id =&gt; &lsquo;protein_id&rsquo;,<br>
transcript_id =&gt; &rsquo;transcript_id&rsquo;,<br>
exon_number =&gt; &rsquo;exon_number&rsquo;,<br>
exon_id =&gt; &rsquo;exon_id&rsquo;,<br>
gene_name =&gt; &lsquo;gene_name&rsquo;);</p>
<p>}<br>
elsif ($method eq &lsquo;source&rsquo;) {<br>
%TAG_MAP = ( organism =&gt; &lsquo;organism&rsquo; );</p>
<p>}<br>
else {</p>
<p>}</p>
<p>my @all_tags = $f-&gt;get_all_tags();<br>
#~ print Dumper \@all_tags;<br>
for my $tag (@all_tags){</p>
<p>if (exists $TAG_MAP{$tag}){<br>
my $newtag= $TAG_MAP{$tag};<br>
my @v= $f-&gt;get_tag_values($tag);<br>
$f-&gt;remove_tag($tag);<br>
$f-&gt;add_tag_value($newtag,@v);<br>
}<br>
else {<br>
$f-&gt;remove_tag($tag);<br>
}<br>
}</p>
<p>return $f;<br>
}</p>
<p>If you would like to access the full document, run:<br>
<code>perldoc NCBI_genbank2gtf.pl</code></p>
<p>Feedback is welcome.</p>
<p>Here is also a GenBank -&gt; BED converter:
<a href="http://bio-spring.top/a-genbank-to-bed-converter/"></a></p>

</div>
</main>

<section class="appendix">







<div>
  <div class="side side-left"><h3>Reuse</h3></div>
  Text and figures are licensed under <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution CC BY 4.0</a>. The source code is licensed under MIT. The full source is available at <a href="https://github.com/yihui/hugo-prose">https://github.com/yihui/hugo-prose</a>.
</div>



<div>
  <div class="side side-left"><h3>Suggest changes</h3></div>
  
  
  
    
    
  
  If you find any mistakes (including typos) or want to suggest changes, please feel free to <a href="https://github.com/yihui/hugo-prose/edit/master/exampleSite/content/post/2015-10-14-a-genbank-gtf-converter-derived-from-bioperl.md" id="edit-link">edit the source file of this page on Github and create a pull request.</a>
</div>




</section>



<nav class="post-nav">
  <span class="nav-next">&larr; <a href="/post/2015/10/14/2015-10-14-a-genbank-to-bed-converter/" title=Next&#32;post&#32;(older)>A Genbank to BED converter</a></span>
  &hercon;
  <span class="nav-prev"><a href="/post/2016/01/12/2016-01-12-contigexpress-assemble-pair-end-sequencing-fragments/" title=Previous&#32;post&#32;(newer)>ContigExpress-Assemble pair-end sequencing fragments</a> &rarr;</span>
</nav>





</div>


  <footer>
  


<script src="//yihui.org/js/math-code.js"></script>
<script async src="//mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML"></script>





<script src="https://cdn.jsdelivr.net/combine/npm/@xiee/utils/js/number-sections.min.js,npm/@xiee/utils/js/toc.min.js,npm/@xiee/utils/js/sidenotes.min.js,npm/@xiee/utils/js/right-quote.min.js,npm/@xiee/utils/js/center-img.min.js,npm/@xiee/utils/js/fix-pandoc.min.js,npm/@xiee/utils/js/heading-anchor.min.js" defer></script>




  <div class="footer">
  
  <ul class="copyright">
    <li>© <a href="https://en.wikipedia.org/wiki/Seneca_the_Younger">Lucius Annaeus Seneca</a> 4 BC &ndash; AD 65</li>
  </ul>
  
  <ul>
    
    <li>
      <a href="/404.html">Contact</a>
    </li>
    
    <li class="optional">
      <a href="/categories/">Categories</a>
    </li>
    
    <li class="optional">
      <a href="/tags/">Tags</a>
    </li>
    
    <li id="menu-edit">
      <a href="#">Suggest an edit</a>
    </li>
    
    <li>
      <a href="#">Back to top</a>
    </li>
    
  </ul>
  </div>
  
  </footer>
  <script src="/js/features.js" defer></script>
  </body>
</html>

